File: add-newsletter.js
// Add newsletter signup + subscribe script to all blog posts
const fs = require('fs');
const path = require('path');

const blogDir = path.join(__dirname, 'blog');

const newsletterHTML = `
    <!-- Newsletter Signup -->
    <div style="background:linear-gradient(135deg,rgba(34,197,94,.06),rgba(34,197,94,.12));border:2px solid rgba(34,197,94,.2);border-radius:16px;padding:2.5rem;margin:3rem 0 2rem;text-align:center">
        <h3 style="font-size:1.4rem;font-weight:800;margin-bottom:.5rem">📬 Get Weekly PM Scaling Tips</h3>
        <p style="color:#64748b;margin-bottom:1.25rem;font-size:.95rem">One actionable tip every week — SOPs, growth strategies, and automation ideas. Join 50+ PM owners. Free forever.</p>
        <form id="blog-nl-form" style="display:flex;gap:.75rem;justify-content:center;flex-wrap:wrap;max-width:500px;margin:0 auto">
            <input type="email" id="blog-nl-email" placeholder="your@email.com" required style="flex:1;min-width:200px;padding:.7rem 1rem;border:2px solid #e2e8f0;border-radius:10px;font-size:.9rem;outline:none" onfocus="this.style.borderColor='#22c55e'" onblur="this.style.borderColor='#e2e8f0'">
            <button type="submit" style="background:#22c55e;color:white;border:none;padding:.7rem 1.5rem;border-radius:10px;font-weight:700;font-size:.9rem;cursor:pointer;white-space:nowrap">Subscribe →</button>
        </form>
        <p id="blog-nl-msg" style="display:none;margin-top:.75rem;font-size:.85rem;font-weight:600"></p>
        <p style="margin-top:.5rem;font-size:.7rem;color:#94a3b8">No spam. Unsubscribe anytime.</p>
    </div>`;

const subscribeScript = `
<script>
document.getElementById('blog-nl-form')?.addEventListener('submit',async function(e){
    e.preventDefault();const email=document.getElementById('blog-nl-email').value;const msg=document.getElementById('blog-nl-msg');const btn=this.querySelector('button');btn.textContent='...';btn.disabled=true;
    try{const r=await fetch('/api/subscribe',{method:'POST',headers:{'Content-Type':'application/json'},body:JSON.stringify({email,source:'blog'})});
    if(r.ok){msg.style.display='block';msg.style.color='#22c55e';msg.textContent='✓ You\\'re in! Weekly tips incoming.';this.style.display='none';}else{throw new Error();}
    }catch(err){msg.style.display='block';msg.style.color='#ef4444';msg.textContent='Error — try again?';btn.textContent='Subscribe →';btn.disabled=false;}
});
</script>`;

let modified = 0;
let skipped = 0;

const files = fs.readdirSync(blogDir).filter(f => f.endsWith('.html'));

for (const file of files) {
    const filePath = path.join(blogDir, file);
    let html = fs.readFileSync(filePath, 'utf8');
    
    // Skip if already has newsletter form
    if (html.includes('blog-nl-form')) {
        skipped++;
        continue;
    }
    
    // Insert newsletter before footer (handle both <div> and <footer> variants)
    let footerIndex = html.indexOf('<div class="footer">');
    if (footerIndex === -1) footerIndex = html.indexOf('<footer class="footer">');
    if (footerIndex === -1) footerIndex = html.indexOf('<footer');
    if (footerIndex === -1) {
        // Last resort: insert before </body>
        footerIndex = html.indexOf('</body>');
    }
    if (footerIndex === -1) {
        console.log(`SKIP (no insertion point): ${file}`);
        skipped++;
        continue;
    }
    
    // Insert newsletter HTML before footer, and script before </body>
    html = html.slice(0, footerIndex) + newsletterHTML + '\n    ' + html.slice(footerIndex);
    html = html.replace('</body>', subscribeScript + '\n</body>');
    
    fs.writeFileSync(filePath, html);
    modified++;
}

console.log(`Done! Modified: ${modified}, Skipped: ${skipped}, Total: ${files.length}`);

File: add-newsletter.py
#!/usr/bin/env python3
"""Add newsletter signup form to blog posts missing it."""
import glob
import os

NEWSLETTER_CSS = """        .newsletter-box{background:#f8f6f0;border:2px solid #f0c040;padding:2rem;border-radius:12px;margin:2rem 0;text-align:center}
        .newsletter-box h3{color:#1a1a2e;margin-bottom:.5rem}
        .newsletter-box p{color:#666;margin-bottom:1rem;font-size:.95rem}
        .newsletter-box form{display:flex;gap:.5rem;justify-content:center;flex-wrap:wrap}
        .newsletter-box input[type="email"]{padding:.6rem 1rem;border:1px solid #ddd;border-radius:6px;font-size:.95rem;min-width:250px}
        .newsletter-box button{background:#f0c040;color:#1a1a2e;border:none;padding:.6rem 1.5rem;border-radius:6px;font-weight:700;cursor:pointer;font-size:.95rem}"""

NEWSLETTER_HTML = """
        <div class="newsletter-box">
            <h3>\U0001f4ec Free Weekly PM Training</h3>
            <p>Get one actionable property management lesson delivered to your inbox every week. No cost, no spam.</p>
            <form id="subscribe-form" onsubmit="return handleSubscribe(event)">
                <input type="email" id="subscribe-email" placeholder="you@example.com" required>
                <button type="submit">Subscribe Free</button>
            </form>
            <p id="subscribe-status" style="margin-top:0.5rem;font-size:0.85rem;"></p>
        </div>
"""

SUBSCRIBE_JS = """    <script>
    async function handleSubscribe(e) {
        e.preventDefault();
        const email = document.getElementById('subscribe-email').value;
        const status = document.getElementById('subscribe-status');
        status.textContent = 'Subscribing...';
        try {
            const res = await fetch('/api/subscribe', {
                method: 'POST',
                headers: {'Content-Type': 'application/json'},
                body: JSON.stringify({email, source: 'blog-newsletter'})
            });
            const data = await res.json();
            if (data.success) {
                status.textContent = '\\u2705 You\\'re in! Check your inbox.';
                status.style.color = '#22c55e';
            } else {
                status.textContent = data.error || 'Something went wrong. Try again.';
                status.style.color = '#ef4444';
            }
        } catch {
            status.textContent = 'Network error. Please try again.';
            status.style.color = '#ef4444';
        }
    }
    </script>
"""

blog_dir = "/home/openclaw/.openclaw/workspaces/agent2/levelpm/blog"
count = 0
errors = []

for filepath in sorted(glob.glob(os.path.join(blog_dir, "*.html"))):
    basename = os.path.basename(filepath)
    if basename == "index.html":
        continue

    with open(filepath, "r") as f:
        content = f.read()

    if "newsletter-box" in content:
        continue

    modified = False

    # 1. Add CSS before </style>
    if "</style>" in content:
        content = content.replace("</style>", NEWSLETTER_CSS + "\n    </style>", 1)
        modified = True

    # 2. Add newsletter HTML before </article>
    if "</article>" in content:
        content = content.replace("</article>", NEWSLETTER_HTML + "    </article>", 1)
        modified = True

    # 3. Add JS before </body>
    if "handleSubscribe" not in content and "</body>" in content:
        content = content.replace("</body>", SUBSCRIBE_JS + "</body>", 1)
        modified = True

    if modified:
        with open(filepath, "w") as f:
            f.write(content)
        count += 1
    else:
        errors.append(basename)

print(f"Newsletter form added to {count} blog posts")
if errors:
    print(f"Skipped (structure issues): {', '.join(errors)}")
